src/operation/insert/mod.rs

```diff
@@ -1,36 +1,37 @@
 #[cfg(test)]
 mod test;
 
-use std::collections::HashMap;
+use std::{collections::HashMap, io::Write};
 
-use bson::{oid::ObjectId, Bson};
+use bson::{oid::ObjectId, spec::ElementType, Bson};
 use serde::Serialize;
 
 use crate::{
-    bson::{doc, Document},
+    bson::doc,
     bson_util,
     cmap::{Command, StreamDescription},
     error::{BulkWriteFailure, Error, ErrorKind, Result},
-    operation::{append_options, Operation, Retryability, WriteResponseBody},
+    operation::{Operation, Retryability, WriteResponseBody},
     options::{InsertManyOptions, WriteConcern},
     results::InsertManyResult,
+    runtime::SyncLittleEndianWrite,
     Namespace,
 };
 
-use super::CommandResponse;
+use super::{CommandBody, CommandResponse};
 
 #[derive(Debug)]
-pub(crate) struct Insert<T> {
+pub(crate) struct Insert<'a, T> {
```

Comment:
Previously this `T` was always actually a reference. This now makes it required to be a reference, which allows us to clone the input.